hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/exceptions/OMNotLeaderException.java

```diff
@@ -34,24 +34,33 @@ public class OMNotLeaderException extends IOException {
 
   private final String currentPeerId;
   private final String leaderPeerId;
+  private final String leaderAddress;
   private static final Pattern CURRENT_PEER_ID_PATTERN =
       Pattern.compile("OM:(.*) is not the leader[.]+.*", Pattern.DOTALL);
   private static final Pattern SUGGESTED_LEADER_PATTERN =
-      Pattern.compile(".*Suggested leader is OM:([^.]*).*", Pattern.DOTALL);
+      Pattern.compile(".*Suggested leader is OM:([^.]*)\\/(.*)\\.",
+          Pattern.DOTALL);
 
   public OMNotLeaderException(RaftPeerId currentPeerId) {
     super("OM:" + currentPeerId + " is not the leader. Could not " +
         "determine the leader node.");
     this.currentPeerId = currentPeerId.toString();
     this.leaderPeerId = null;
+    this.leaderAddress = null;
   }
 
   public OMNotLeaderException(RaftPeerId currentPeerId,
       RaftPeerId suggestedLeaderPeerId) {
+    this(currentPeerId, suggestedLeaderPeerId, null);
+  }
+
+  public OMNotLeaderException(RaftPeerId currentPeerId,
+      RaftPeerId suggestedLeaderPeerId, String suggestedLeaderAddress) {
     super("OM:" + currentPeerId + " is not the leader. Suggested leader is" +
-        " OM:" + suggestedLeaderPeerId + ".");
+        " OM:" + suggestedLeaderPeerId + "/" + suggestedLeaderAddress + ".");
```

Comment:
Thanks @symious for this important patch to fix suggesting the leader om when the client submits a request to a follower.  
Q. Will we also run into the client nodeid list misconfiguration problem brought up should the `OMNotALeaderException `return a `null` for the `suggestedLeaderAddress`?

> In https://github.com/apache/ozone/pull/2765#issuecomment-952091699, the concern I think is the misconfig of client side might trigger some dead loops,

Reply:
@neils-dev Thank you for the review.
I think the `suggestedLeaderAddress` will return `null` if there are some errors on the server side about the Ratis Peer or hostname resolving, which should be resolved on server side according to the error logs. Once the server side error fix, the client should be able to realise the error config according to the exception message.

Reply:
Added a check if the Server and Client information of OM matches, if not, it will fallback to the initial implementation.